src/core/main.c

```diff
@@ -273,6 +273,14 @@ _noreturn_ static void crash(int sig) {
                         siginfo_t status;
                         int r;
 
+                        if (siginfo) {
+                                _cleanup_free_ char *cmdline = NULL;
+                                pid_t sender_pid = siginfo->si_pid;
+
+                                (void) get_process_cmdline(sender_pid, SIZE_MAX, 0, &cmdline);
```

Comment:
this calls malloc internally. We so far avoided doing that from the crash handler, since there's a good chance it is called due to some memory corruption issue, where we shouldn't do more of it, since it likely will fail again, even more catastrophically.

I mean, as a general rule, memory allocations from signal handlers are *not* OK.

hence it might make sense to show the PID of the process that killed us, but I'd avoid anything with reading the cmdline, here, it's not really worth the effrot.

i.e. drop the cmdline output, leave the PID part in.

Reply:
I thought about the malloc thing, but given it is used best-effort (failure simply means it shows `n/a`), I thought it would be fine.

Reply:
well, if glibc sees a memory corruption while allocating memory it will hit an assert. We should risk that in our crash handler, which likely is called when memory corruption already took place.

malloc() is generally not safe to be called from signal handlers, but particularly not in case that signal handler is for SIGSEGV and friends.

Reply:
ok, sounds good to me